libc/unistd_getopt.c

```diff
@@ -0,0 +1,337 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Testing POSIX compatibility of getopt() unistd.h function
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Marek Bialowas
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <ctype.h>
+
+#include <unity_fixture.h>
+
+
+enum getopterr { NOERR = 0,
+	OPTREQ = 1,
+	UNKNOWNOPT = 2,
+	UNKNOWNCHAR = 4,
+	NOARG = 8,
+	ABORT = 16 };
+
+
+struct getopt_t {
+	int aflag;
+	int bflag;
+	char *cvalue;
+	int nonopts;
+
+	enum getopterr err;
+};
+
+
+int testargc = 0;
+struct getopt_t ret;
+
+/* updates `results` structure according to getopt() operations */
+int testmain(int argc, char *argv[], struct getopt_t *results, const char *optstring)
+{
+	int c;
+	int index;
```

Comment:
Oops - rename
https://stackoverflow.com/questions/26515237/whenever-i-use-index-as-global-variable-i-get-an-error

Reply:
Thanks for pointing it out. Changed to `arg`.